src/Files.App/Helpers/AppLifecycleHelper.cs

```diff
@@ -0,0 +1,302 @@
+﻿using Files.App.Services.Settings;
+using Microsoft.Windows.AppLifecycle;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Text.Json;
+using Files.App.Helpers;
+
+namespace Files.App.Helpers
+{
+	public static class AppLifecycleHelper
+	{
+		private const long defaultBufferSize = 1024;
+
+		private const string sharedMemoryHeaderName = "FilesAppTabsWithID";
+		private static MemoryMappedFile? sharedMemoryHeader;
+
+		private static string defaultSharedMemoryName = sharedMemoryHeaderName + defaultBufferSize.ToString();
+		private static string sharedMemoryName = defaultSharedMemoryName;
+		public static string instanceId = Process.GetCurrentProcess().Id.ToString();
+		private static MemoryMappedFile? sharedMemory;
+
+		private static IUserSettingsService userSettingsService = Ioc.Default.GetRequiredService<IUserSettingsService>();
+		private static List<TabItemWithIDArguments> tabsWithIdArgList = new List<TabItemWithIDArguments>();
+		/* 
+		Add sharedMemoryHeaderName because sharedMemory can't be released instantly and create a new one with the same name.
+		To dynamically expand the size of a sharedMemory, a new sharedMemory with a new name needs to be created.
+		Using sharedMemoryHeader, sharedMemoryName is saved and shared between all the Files instance.
+		*/
+		// Read or create a sharedMemory(sharedMemoryHeaderName) to save and share the sharedMemoryName
+		// The sharedMemoryName is used to identify the sharedMemory
+		// The sharedMemory is used to save and share the TabItemWithIDArguments
+		private static MemoryMappedFile GetSharedMemoryNameHeader()
+		{
+			try
+			{
+				sharedMemoryHeader = MemoryMappedFile.OpenExisting(sharedMemoryHeaderName);
+				using (var accessor = sharedMemoryHeader.CreateViewAccessor())
+				{
+					var buffer = new byte[accessor.Capacity];
+					accessor.ReadArray(0, buffer, 0, buffer.Length);
+					var nullIndex = Array.IndexOf(buffer, (byte)'\0');
+					if (nullIndex > 0)
+					{
+						var truncatedBuffer = new byte[nullIndex];
+						Array.Copy(buffer, 0, truncatedBuffer, 0, truncatedBuffer.Length);
+						sharedMemoryName = Encoding.UTF8.GetString(truncatedBuffer);
+					}
+					else
+					{
+						sharedMemoryName = defaultSharedMemoryName;
+					}
+				}
+			}
+			catch (FileNotFoundException)
+			{
+				sharedMemoryName = defaultSharedMemoryName;
+				sharedMemoryHeader = MemoryMappedFile.CreateOrOpen(sharedMemoryHeaderName, defaultBufferSize);
+				using (var accessor = sharedMemoryHeader.CreateViewAccessor())
+				{
+					byte[] buffer = Encoding.UTF8.GetBytes(sharedMemoryName);
+					accessor.WriteArray(0, buffer, 0, buffer.Length);
+				}
+			}
+			return sharedMemoryHeader;
+		}
+
+		//	Check if the sharedMemory exists, if not create it
```

Comment:
Please add [XML comments](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags), which enables us to generate docs easily instead of single line comment.